test/EndToEnd/tests/BuildIntegratedTest.ps1

```diff
@@ -588,4 +588,22 @@ function Test-BuildIntegratedLegacyRebuildDoesNotDeleteCacheFile {
 
     #Assert
```

Comment:
I would also suggest adding an LSL test for the same scenario. 
I did some testing myself, so I used the following test. I would add it to LightweightSolutionLoadTest.ps1

```
function Test-DeferredBuildIntegratedGetIncompatibleError {
    [SkipTestForVS14()]
      
    $projectR = New-Project Net45BuildIntegratedClassLibrary #project.json
    $projectT = New-Project PackageReferenceClassLibrary #this is a net46 library, incompatible

    $projectR | Add-ProjectReference -ProjectTo $projectT

    Clean-Solution

    $projectR = $projectR | Select-Object UniqueName, ProjectName, FullName

    Enable-LightweightSolutionLoad -Reload

    # Act (Restore)
    Build-Solution

    Assert-True ($projectR | Test-Project -IsDeferred) -Message 'Test project should stay in deferred mode'
    
    Assert-ProjectJsonLockFileErrorCode $projectR NU1201
}
```